src/core/dash.h

```diff
@@ -372,16 +372,15 @@ class DashTable<_Key, _Value, Policy>::Iterator {
     return *this;
   }
 
-  detail::IteratorPair<Key_t, Value_t> operator->() {
+  auto operator->() const {
```

Comment:
the changes here is that you return `const Key_t` if the operator is const?

Reply:
I'd prefer to see 2 implementations for const and non const operator

Reply:
> the changes here is that you return `const Key_t` if the operator is const?

Yes, but more importantly `const Value_t`

> I'd prefer to see 2 implementations for const and non const operator

There's a single `const` implementation, the non-const one (which is removed) was not in use..

Reply:
I mean if we use const operator that gives us ability to modify something it's a little incosistent

Reply:
I agree that it's inconsistent, but it's the existing code and beyond the scope of my changes :)

Reply:
> I mean if we use const operator that gives us ability to modify something it's a little incosistent

> I agree that it's inconsistent

Why? It really depends on how you treat the types, especially non owning or views, and what safety you want. I don't see anything wrong with it with iterators. In our case the iterator itself provides conditional mutability based on its own type

For example std::vector::iterator does the same:
```c++
vector<int> v = {1,2,3};
const auto it = v.begin();
*it = 11; // works even if `it` is "const", and cbegin() wouldn't even if mutable
```

Reply:
It depends on how you look at it, but I think a better example would be:

```cpp
    vector<int> v = {1,2,3};
    const vector<int>& vr = v;
    auto it = vr.begin();
    *it = 1;  // Does not compile
```

As in, a `const` method returning a (`const`) iterator should not allow mutating data pointer to by the iterator...

Reply:
(but again, beyond the scope of this PR)